packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -679,6 +734,107 @@ const Headers = ({ titles, classes, headerClasses }: { titles: string[], headerC
 
 export type ExtendedDialogueMatchPreferenceTopic = DbDialogueMatchPreference["topicPreferences"][number] & {matchedPersonPreference?: "Yes" | "Meh" | "No", recommendationReason: string, theirVote?: string, yourVote?: string}
 
+type UserRecommendedContentProps = {
+  classes: ClassesType<typeof styles>;
+  targetUserId: string;
+}
+
+const UserRecommendedContent = ({ classes, targetUserId }: UserRecommendedContentProps ) => {
+  const { PostsTooltip, Loading, CommentView, TopicSuggestion, ExpandCollapseText } = Components;
+
+  const [isExpanded, setIsExpanded] = useState(false);
+  const currentUser = useCurrentUser();
+  const { captureEvent } = useTracking()
+
+  const { loading: tagLoading, error: tagError, data: tagData } = useQuery(gql`
+  query UserTopTags($userId: String!) {
+    UserTopTags(userId: $userId) {
+      tag {
+        name
+        _id
+      }
+      commentCount
+    }
+  }
+`, {
+  variables: { userId: targetUserId },
+  skip: !currentUser
+});
+
+const { loading: postsLoading, error: postsError, data: postsData } = useQuery(gql`
+  query UsersReadPostsOfTargetUser($userId: String!, $targetUserId: String!, $limit: Int!) {
+    UsersReadPostsOfTargetUser(userId: $userId, targetUserId: $targetUserId, limit: $limit) {
+      _id
+      title
+      slug
+    }
+  }
+`, {
+  variables: { userId: currentUser?._id, targetUserId: targetUserId, limit : 4 },
+  skip: !currentUser
+});
+
+const { loading: commentsLoading, error: commentsError, data: commentsData } = useQuery(gql`
+  query UsersRecommendedCommentsOfTargetUser($userId: String!, $targetUserId: String!, $limit: Int!) {
+    UsersRecommendedCommentsOfTargetUser(userId: $userId, targetUserId: $targetUserId, limit: $limit) {
+      _id
+      postId
+      contents {
+        html
+        plaintextMainText
+      }
+    }
+  }
+`, {
+  variables: { userId: currentUser?._id, targetUserId: targetUserId, limit : 3 },
+  skip: !currentUser
+});
+
+const topTags:TagWithCommentCount[] | undefined = tagData?.UserTopTags;
+const readPosts:PostYouveRead[] | undefined = postsData?.UsersReadPostsOfTargetUser
+const recommendedComments:RecommendedComment[] | undefined = commentsData?.UsersRecommendedCommentsOfTargetUser
+
+if (!currentUser || !topTags || !readPosts || !recommendedComments) return <Loading />;
```

Comment:
Is this (and most of the preceeding logic) copied from somewhere?  Looks very familiar

Reply:
it is. Considered putting it into a reusable hook, but, third times the charm